gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -89,6 +88,112 @@ def _load_cadd_raw(cadd_tsv) -> hl.Table:
+    :param ensembl_ids: Path to Ensembl 105 ID file, downloaded from Ensembl 105 archive.
+    :return: Hail Table with REVEL scores for GRCh38.
+    """
+    ht = hl.import_table(
+        revel_csv,
+        delimiter=",",
+        min_partitions=1000,
+        types={"grch38_pos": hl.tstr, "REVEL": hl.tfloat64},
+    )
+    ht = ht.checkpoint(
+        "gs://gnomad-tmp-4day/revel-v1.3_all_chromosomes_with_transcript_ids.ht",
+        _read_if_exists=True,
+    )
+
+    logger.info("Annotating REVEL table...")
+    ht = ht.drop("hg19_pos", "aaref", "aaalt")
+    ht = ht.filter(ht.grch38_pos.contains("."), keep=False)
+    ht = ht.transmute(chr="chr" + ht.chr)
+    ht = ht.annotate(
+        locus=hl.locus(ht.chr, hl.int(ht.grch38_pos), reference_genome="GRCh38"),
+        alleles=hl.array([ht.ref, ht.alt]),
+        Transcript_stable_ID=ht.Ensembl_transcriptid.strip().split(";"),
+    )
+    ht = ht.select("locus", "alleles", "REVEL", "Transcript_stable_ID")
+    ht = ht.explode("Transcript_stable_ID")
+    ht = ht.key_by("Transcript_stable_ID")
+
+    logger.info("Import and process the ensembl ID file...")
+    ensembl_ids = hl.import_table(ensembl_ids, min_partitions=200, impute=True)
+    ensembl_ids = ensembl_ids.select(
+        "Transcript_stable_ID", "Ensembl_Canonical", "MANE_Select"
+    )
+    ensembl_ids = ensembl_ids.key_by("Transcript_stable_ID")
+
+    ht = ht.annotate(
+        in_Ensembl105=hl.is_defined(ensembl_ids[ht.Transcript_stable_ID]),
+        MANE_Select=ensembl_ids[ht.Transcript_stable_ID].MANE_Select,
+        Ensembl_Canonical=ensembl_ids[ht.Transcript_stable_ID].Ensembl_Canonical,
+    )
+
+    logger.info("Filtering transcripts that are in Ensembl 105...")
+    ht = ht.filter(hl.is_defined(ht.in_Ensembl105), keep=True)
+
+    logger.info(
+        "Annotating REVEL scores for MANE Select transcripts and canonical"
+        " transcripts..."
+    )
+    ht = ht.key_by("locus", "alleles")
+    ht = ht.annotate(
+        revel_mane=hl.if_else(ht.MANE_Select != "", ht.REVEL, hl.missing(hl.tfloat)),
+        revel_canonical=hl.if_else(
+            ht.Ensembl_Canonical == "1", ht.REVEL, hl.missing(hl.tfloat)
+        ),
+        revel_noncanonical=hl.if_else(
+            ht.Ensembl_Canonical != "1", ht.REVEL, hl.missing(hl.tfloat)
+        ),
+    )
+
+    logger.info("Taking max REVEL scores for MANE Select transcripts...")
+    ht = ht.checkpoint(
+        "gs://gnomad-tmp-4day/revel-v1.3_in_Ensembl105.ht", _read_if_exists=True
+    )
+    mane_ht = ht.filter(hl.is_defined(ht.revel_mane), keep=True)
+    max_revel_mane = mane_ht.group_by(*mane_ht.key).aggregate(
+        max_revel=hl.agg.max(mane_ht.revel_mane)
+    )
+    transcripts_mane = mane_ht.group_by(*mane_ht.key).aggregate(
+        transcripts=hl.agg.collect_as_set(mane_ht.Transcript_stable_ID)
+    )
+    max_revel_mane = max_revel_mane.annotate(
+        transcripts=transcripts_mane[max_revel_mane.key].transcripts
+    )
```

Comment:
```suggestion
    max_revel_mane = mane_ht.group_by(*mane_ht.key).aggregate(
        max_revel=hl.agg.max(mane_ht.revel_mane), 
        transcripts=hl.agg.collect_as_set(mane_ht.Transcript_stable_ID),
        )
```
I think this does the same thing you are